contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -272,31 +326,59 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
         _rebalance(_amountToSwap, _minTokenReceived, _swapWETH);
     }
 
+    /// @notice Only used for the initial deposit (or after calling withdrawAll) when there is no liquidity
+    ///         in the [-1, 0] tick. This function is needed since we can not swap to the desired position within
+    ///         the pool if there is no liquidity in the [-1, 0] ticker
+    function depositLiquidity() external nonReentrant onlyGovernorOrStrategist {
+        require(tokenId == 0, "Liquidity already deposited");
+        _addLiquidity();
+    }
+
     /**
      * @dev Rebalance the pool to the desired token split
      */
     function _rebalance(uint256 _amountToSwap, uint256 _minTokenReceived, bool _swapWETH) internal {
-        _removeLiquidity();
+        _removePartialLiquidity(withdrawLiquidityShare);
         _swapToDesiredPosition(_amountToSwap, _minTokenReceived, _swapWETH);
+        // calling check liquidity early so we don't get unexpected errors when adding liquidity
+        // in the later stages of this function
+        _checkLiquidityWithinExpectedShare();
         _addLiquidity();
         _checkLiquidityWithinExpectedShare();
     }
 
     /**
-     * @dev Decrease withdrawLiquidityShare (currently set to 99%) of the liquidity if strategy holds any. In practice the removal of liquidity
-     * will be skipped only on the first time called.
+     * @dev Decrease partial liquidity from the pool.
+     * @param _partialLiquidityToDecrease The amount of liquidity to remove expressed in basis points
      */
-    function _removeLiquidity() internal {
-        if (tokenId == 0) {
-            return;
-        }
+    function _removePartialLiquidity(uint256 _partialLiquidityToDecrease) internal {
```

Comment:
Using BPS here probably isn't the best solution. Using something 1e18 denominated gives us higher resolution, and doesn't add an unusual conversion factor. In general I think we should only use bps when we really need to pack some data into a slot.

Reply:
fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/a6e6ca721b817187694d65a793c04e7760a02666